pkg/kube/util.go

```diff
@@ -164,16 +164,17 @@ func CheckPodReady(pod *kubeApiCore.Pod) error {
 }
 
 // GetDeployMetaFromPod heuristically derives deployment metadata from the pod spec.
-func GetDeployMetaFromPod(pod *kubeApiCore.Pod) (*metav1.ObjectMeta, *metav1.TypeMeta) {
+func GetDeployMetaFromPod(pod *kubeApiCore.Pod) (metav1.ObjectMeta, metav1.TypeMeta) {
 	if pod == nil {
-		return nil, nil
+		return metav1.ObjectMeta{}, metav1.TypeMeta{}
 	}
 	// try to capture more useful namespace/name info for deployments, etc.
 	// TODO(dougreid): expand to enable lookup of OWNERs recursively a la kubernetesenv
-	deployMeta := pod.ObjectMeta.DeepCopy()
-	deployMeta.Namespace = pod.ObjectMeta.Namespace
+	deployMeta := pod.ObjectMeta
```

Comment:
do we still need the `deployMeta.Namespace = pod.ObjectMeta.Namespace`?

Reply:
In k8s, I can not think of a case where they are different.

Reply:
good point